docs/index.bs

```diff
@@ -3100,6 +3122,21 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
   </section>
 
   <section algorithm>
+    <h3 id="update-service-worker-extended-events-set-algorithm"><dfn>Update Service Worker Extended Events Set</dfn></h3>
```

Comment:
Good point! I added a step to Terminate Service Worker to remove all the items in the service worker's set of extended events.

For extended events, I thought we decided to leave them run to completion: https://github.com/w3c/ServiceWorker/issues/916#issuecomment-229332311. So, I removed the steps that abort the in-progress requests.

To clarify what happens when a service worker abruptly terminates, Terminate Service Worker resets the set of extended events to an empty set (done in this PR), retains the tasks (for functional events) in the queue so the next worker run will execute them.